pygsuite/sheets/cell.py

```diff
@@ -0,0 +1,127 @@
+from dataclasses import dataclass
+from typing import Dict, Optional, Tuple
+
+from pygsuite.common.style import Color, TextStyle
+
+
+@dataclass
+class CellFormat:
+
+    HORIZONTAL_ALIGNMENTS = ["LEFT", "CENTER", "RIGHT"]
+    VERTICAL_ALIGNMENTS = ["TOP", "MIDDLE", "BOTTOM"]
+    WRAP_STRATEGIES = ["OVERFLOW_CELL", "LEGACY_WRAP", "CLIP", "WRAP"]
+    TEXT_DIRECTIONS = ["LEFT_TO_RIGHT", "RIGHT_TO_LEFT"]
+    HYPERLINK_DISPLAY_TYPES = ["LINKED", "PLAIN_TEXT"]
+
+    number_format: Optional[dict] = None
+    background_color: Optional[Color] = None
+    background_color_style: Optional[dict] = None
+    borders: Optional[dict] = None
+    padding: Optional[dict] = None
+    horizontal_alignment: Optional[str] = None
+    vertical_alignment: Optional[str] = None
+    wrap_strategy: Optional[str] = None
+    text_direction: Optional[str] = None
+    text_format: Optional[TextStyle] = None
+    hyperlink_display_type: Optional[str] = None
+    text_rotation: Optional[dict] = None
+
+    def to_json(self) -> Tuple[list, Dict]:  # noqa: C901
+
+        base = {}
+        fields_mask = []
+
+        if self.number_format is not None:
+            base["numberFormat"] = self.number_format
+
+        if self.background_color is not None:
+            base["backgroundColor"] = self.background_color
+
+        if self.background_color_style is not None:
+            base["backgroundColorStyle"] = self.background_color_style
+
+        if self.borders is not None:
```

Comment:
there's got to be a better way for us to do all of these but (less repetitive) but definetly fine, haven't come up with any great ideas.